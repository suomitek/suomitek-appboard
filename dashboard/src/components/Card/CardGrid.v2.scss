.card-grid {
  padding-right: 1.2rem;
  padding-left: 1.2rem;
}
