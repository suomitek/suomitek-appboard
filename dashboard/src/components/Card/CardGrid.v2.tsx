import * as React from "react";
import Row from "../js/Row";

import "./CardGrid.v2.css";

function CardGrid(props: { children: JSX.Element | JSX.Element[] }) {
  return (
    <div className="card-grid">
      <Row>{props.children}</Row>
    </div>
  );
}

export default CardGrid;
