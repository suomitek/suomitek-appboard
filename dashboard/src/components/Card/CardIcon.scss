.Card__icon {
  display: flex;
  height: calc(72px + 2em);
  align-items: center;
  justify-content: center;
}

.Card__icon img {
  max-height: 72px;
  max-width: 160px;
}

.Card__subIcon {
  right: 10px;
  position: relative;
  bottom: 27px;
  background: transparent;
  height: 0px;
}

.Card__subIcon img {
  max-height: 25px;
  max-width: 25px;
}
