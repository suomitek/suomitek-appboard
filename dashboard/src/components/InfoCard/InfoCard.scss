.ListItem {
  color: #1c2b39;
}

.ListItem__content__title {
  margin: 0;
  width: 100%;
  font-weight: bold;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  color: #1c2b39;
}

.ListItem__content__info {
  display: flex;
  justify-content: space-between;
}

.ListItem__content__info_tag {
  float: right;
  margin-left: 3px;
  margin-bottom: 3px;
  color: #fff;
  border-radius: 16px;
  max-width: 120px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.ListItem__content__info_tag-2 {
  background-color: #1c2b39;
}

.ListItem__content__info_tag-1 {
  background-color: #1598cb;
}

.ListItem__header {
  position: relative;
  display: block;
}

.ListItem__banner {
  position: absolute;
  bottom: 0;
  background: #008145;
  color: white;
  width: 100%;
  left: 0;
  margin: 0;
  text-align: center;
  padding: 0.25em 0;
  font-size: 14px;
}
