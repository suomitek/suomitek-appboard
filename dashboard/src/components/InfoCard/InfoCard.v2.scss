.card-header {
  .card-title {
    max-width: calc(100% - 24px);
    overflow-x: hidden;
    text-overflow: ellipsis;
  }
  img {
    position: absolute;
    max-width: 24px;
    top: 14px;
    right: 1rem;
  }
}

.card-block {
  height: 4.5rem;
  overflow-y: hidden;
  .card-icon {
    display: flex;
    align-items: center;
    text-align: center;
    height: 3.5rem;
    img {
      max-width: 100%;
    }
  }
}

.kubeapps-card-footer {
  display: flex;
  justify-content: space-between;
  width: 100%;
  .kubeapps-card-footer-col1 {
    width: 100%;
    margin: 0 0.6rem 0 0.6rem;
  }
  .kubeapps-card-footer-col2 {
    margin: 0 0.5rem 0 0;
  }
}

.card.clickable a {
  text-decoration: none;
  color: black;
}

.footer-tags {
  text-align: right;
}
