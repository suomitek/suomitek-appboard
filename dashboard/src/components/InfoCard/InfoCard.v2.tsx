import * as React from "react";
import { Link } from "react-router-dom";
import Card, { CardBlock, CardFooter, CardHeader } from "../js/Card";
import Column from "../js/Column";
import Row from "../js/Row";

import placeholder from "../../placeholder.png";
import "./InfoCard.v2.css";

export interface IInfoCardProps {
  title: string;
  info: string | JSX.Element;
  link?: string;
  icon?: string;
  subIcon?: string;
  description?: string | JSX.Element;
  tag1Class?: string;
  tag1Content?: string | JSX.Element;
  tag2Class?: string;
  tag2Content?: string | JSX.Element;
}

function InfoCard(props: IInfoCardProps) {
  const {
    title,
    link,
    info,
    description,
    tag1Content,
    tag1Class,
    tag2Content,
    tag2Class,
    subIcon,
  } = props;
  const icon = props.icon ? props.icon : placeholder;
  return (
    <Column span={[12, 6, 4, 3]}>
      <Card clickable={true}>
        <Link to={link || "#"}>
          <CardHeader>
            <>
              <div className="card-title">{title}</div>
              {subIcon && <img src={subIcon} alt="icon" />}
            </>
          </CardHeader>
          <CardBlock>
            <Row>
              <Column span={3}>
                <div className="card-icon">
                  <img src={icon} alt="icon" sizes="64px" />
                </div>
              </Column>
              <Column span={9}>
                <span>{description}</span>
              </Column>
            </Row>
          </CardBlock>
          <CardFooter>
            <Row>
              <div className="kubeapps-card-footer">
                <div className="kubeapps-card-footer-col1">{info}</div>
                <div className="kubeapps-card-footer-col2">
                  <div className="footer-tags">
                    {tag1Content && (
                      <span className={`label ${tag1Class || "label-info"}`}>{tag1Content}</span>
                    )}
                    {tag2Content && (
                      <span className={`label ${tag2Class || "label-info"}`}>{tag2Content}</span>
                    )}
                  </div>
                </div>
              </div>
            </Row>
          </CardFooter>
        </Link>
      </Card>
    </Column>
  );
}

export default InfoCard;
