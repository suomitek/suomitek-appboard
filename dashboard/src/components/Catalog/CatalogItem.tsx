import * as React from "react";
import { useSelector } from "react-redux";
import { Link } from "react-router-dom";
import helmIcon from "../../icons/helm.svg";
import operatorIcon from "../../icons/operator-framework.svg";
import placeholder from "../../placeholder.png";
import { IRepo, IStoreState } from "../../shared/types";
import * as url from "../../shared/url";
import InfoCard from "../InfoCard";
import "./CatalogItem.css";

export interface ICatalogItem {
  id: string;
  name: string;
  version: string;
  description: string;
  cluster: string;
  namespace: string;
  icon?: string;
}

export interface IChartCatalogItem extends ICatalogItem {
  repo: IRepo;
}

export interface IOperatorCatalogItem extends ICatalogItem {
  csv: string;
}

export interface ICatalogItemProps {
  type: string;
  item: IChartCatalogItem | IOperatorCatalogItem;
}

// 3 lines description max
const MAX_DESC_LENGTH = 90;

function trimDescription(desc: string): string {
  if (desc.length > MAX_DESC_LENGTH) {
    // Trim to the last word under the max length
    return desc.substr(0, desc.lastIndexOf(" ", MAX_DESC_LENGTH)).concat("...");
  }
  return desc;
}

const CatalogItem: React.SFC<ICatalogItemProps> = props => {
  if (props.type === "operator") {
    const item = props.item as IOperatorCatalogItem;
    return OperatorCatalogItem(item);
  } else {
    const item = props.item as IChartCatalogItem;
    return ChartCatalogItem(item);
  }
};

const OperatorCatalogItem: React.SFC<IOperatorCatalogItem> = props => {
  const { icon, name, csv, version, description, cluster, namespace, id } = props;
  const iconSrc = icon || placeholder;
  // Cosmetic change, remove the version from the csv name
  const csvName = props.csv.split(".v")[0];
  const tag1 = <span>{csvName}</span>;
  const link = url.app.operatorInstances.new(cluster, namespace, csv, id);
  const subIcon = operatorIcon;
  const descriptionC = (
    <div className="ListItem__content__description">{trimDescription(description)}</div>
  );
  return (
    <InfoCard
      key={id}
      title={name}
      link={link}
      info={version || "-"}
      icon={iconSrc}
      description={descriptionC}
      tag1Content={tag1}
      tag1Class={""}
      subIcon={subIcon}
    />
  );
};

const ChartCatalogItem: React.SFC<IChartCatalogItem> = props => {
  const { icon, name, repo, version, description, namespace, id } = props;
  const iconSrc = icon || placeholder;
  const cluster = useSelector((state: IStoreState) => state.clusters.currentCluster);
  const tag1 = (
    <Link
      className="ListItem__content__info_tag_link"
      to={url.app.repo(cluster, namespace, repo.name)}
    >
      {repo.name}
    </Link>
  );
  const link = url.app.charts.get(cluster, namespace, name, repo || ({} as IRepo));
  const subIcon = helmIcon;

  const descriptionC = (
    <div className="ListItem__content__description">{trimDescription(description)}</div>
  );
  return (
    <InfoCard
      key={id}
      title={name}
      link={link}
      info={version || "-"}
      icon={iconSrc}
      description={descriptionC}
      tag1Content={tag1}
      tag1Class={repo ? repo.name : ""}
      subIcon={subIcon}
    />
  );
};

export default CatalogItem;
