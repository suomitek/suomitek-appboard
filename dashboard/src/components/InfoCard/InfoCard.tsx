import * as React from "react";

import { Link } from "react-router-dom";
import placeholder from "../../placeholder.png";
import Card, { CardContent, CardIcon } from "../Card";
import "./InfoCard.css";

export interface IServiceInstanceCardProps {
  title: string;
  info: string;
  link?: string;
  icon?: string;
  banner?: string;
  subIcon?: string;
  description?: string | JSX.Element;
  tag1Class?: string;
  tag1Content?: string | JSX.Element;
  tag2Class?: string;
  tag2Content?: string | JSX.Element;
}

const InfoCard: React.SFC<IServiceInstanceCardProps> = props => {
  const {
    title,
    link,
    info,
    description,
    tag1Content,
    tag1Class,
    tag2Content,
    tag2Class,
    banner,
    subIcon,
  } = props;
  const icon = props.icon ? props.icon : placeholder;
  return (
    <Card responsive={true} className="ListItem">
      <Link to={link || "#"} title={title} className="ListItem__header">
        <CardIcon icon={icon} />
        {banner && <p className="ListItem__banner">{banner}</p>}
        {subIcon && (
          <div className="Card__subIcon bg-light text-r">
            <img src={subIcon} alt="icon" />
          </div>
        )}
      </Link>
      <CardContent>
        <div className="ListItem__content">
          <Link to={link || "#"} title={title}>
            <h3 className="ListItem__content__title type-big">{title}</h3>
          </Link>
          {description}
          <div className="ListItem__content__info">
            <p className="margin-reset type-small padding-t-tiny type-color-light-blue">{info}</p>
            <div>
              {tag1Content && (
                <span
                  className={`ListItem__content__info_tag ListItem__content__info_tag-1 type-small type-color-white padding-t-tiny padding-h-normal ${tag1Class ||
                    ""}`}
                >
                  {tag1Content}
                </span>
              )}
              {tag2Content && (
                <span
                  className={`ListItem__content__info_tag ListItem__content__info_tag-2 type-small type-color-white padding-t-tiny padding-h-normal margin-b-normal ${tag2Class ||
                    ""}`}
                >
                  {tag2Content}
                </span>
              )}
            </div>
          </div>
        </div>
      </CardContent>
      {props.children}
    </Card>
  );
};

export default InfoCard;
