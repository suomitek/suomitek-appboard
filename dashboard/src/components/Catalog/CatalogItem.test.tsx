import { shallow } from "enzyme";
import context from "jest-plugin-context";
import * as React from "react";
import { getStore, mountWrapper } from "shared/specs/mountWrapper";

import { IRepo } from "../../shared/types";
import { CardIcon } from "../Card";
import InfoCard from "../InfoCard";
import CatalogItem, {
  ICatalogItemProps,
  IChartCatalogItem,
  IOperatorCatalogItem,
} from "./CatalogItem";

jest.mock("../../placeholder.png", () => "placeholder.png");

const defaultItem = {
  id: "foo1",
  name: "foo",
  version: "1.0.0",
  description: "",
  type: "chart",
  repo: {
    name: "repo-name",
    namespace: "repo-namespace",
  } as IRepo,
  cluster: "default",
  namespace: "repo-namespace",
  icon: "icon.png",
} as IChartCatalogItem;

const defaultProps: ICatalogItemProps = {
  item: defaultItem,
  type: "chart",
};

const defaultStore = getStore({});

it("should render a chart item in a namespace", () => {
  const wrapper = mountWrapper(defaultStore, <CatalogItem {...defaultProps} />);
  // Can't shallow render connected components for easy snapshotting :/
  // https://github.com/enzymejs/enzyme/issues/2202
  expect(wrapper.find(InfoCard)).toMatchSnapshot();
});

it("should render a global chart item in a namespace", () => {
  const props = {
    ...defaultProps,
    item: {
      ...defaultItem,
      repo: {
        name: "repo-name",
        namespace: "kubeapps",
      } as IRepo,
    },
  };
  const wrapper = mountWrapper(defaultStore, <CatalogItem {...props} />);
  expect(wrapper.find(InfoCard)).toMatchSnapshot();
});

it("should use the default placeholder for the icon if it doesn't exist", () => {
  const props = {
    ...defaultProps,
    item: {
      ...defaultItem,
      icon: undefined,
    },
  };
  const wrapper = mountWrapper(defaultStore, <CatalogItem {...props} />);
  // Importing an image returns "undefined"
  expect(wrapper.find(CardIcon).prop("src")).toBe(undefined);
});

it("should place a dash if the version is not avaliable", () => {
  const props = {
    ...defaultProps,
    item: {
      ...defaultItem,
      version: "",
    },
  };
  const wrapper = mountWrapper(defaultStore, <CatalogItem {...props} />);
  expect(wrapper.find(".type-color-light-blue").text()).toBe("-");
});

it("show the chart description", () => {
  const props = {
    ...defaultProps,
    item: {
      ...defaultItem,
      description: "This is a description",
    },
  };
  const wrapper = mountWrapper(defaultStore, <CatalogItem {...props} />);
  expect(wrapper.find(".ListItem__content__description").text()).toBe(props.item.description);
});

context("when the description is too long", () => {
  it("trims the description", () => {
    const props = {
      ...defaultProps,
      item: {
        ...defaultItem,
        description:
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum ultrices velit leo, quis pharetra mi vestibulum quis.",
      },
    };
    const wrapper = mountWrapper(defaultStore, <CatalogItem {...props} />);
    expect(wrapper.find(".ListItem__content__description").text()).toMatch(/\.\.\.$/);
  });
});

context("when the item is a catalog", () => {
  const catalogItem = {
    ...defaultItem,
    csv: "foo-cluster",
  } as IOperatorCatalogItem;

  it("shows the proper tag", () => {
    const wrapper = shallow(<CatalogItem item={catalogItem} type={"operator"} />);
    expect((wrapper.find(InfoCard).prop("tag1Content") as JSX.Element).props.children).toEqual(
      "foo-cluster",
    );
  });

  it("has the proper link", () => {
    const wrapper = shallow(<CatalogItem item={catalogItem} type={"operator"} />);
    expect(wrapper.find(InfoCard).prop("link")).toEqual(
      `/c/default/ns/${defaultItem.namespace}/operators-instances/new/foo-cluster/foo1`,
    );
  });
});
