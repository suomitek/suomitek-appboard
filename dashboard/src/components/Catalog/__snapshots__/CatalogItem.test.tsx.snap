// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a chart item in a namespace 1`] = `
<InfoCard
  description={
    <div
      className="ListItem__content__description"
    >
      
    </div>
  }
  icon="icon.png"
  info="1.0.0"
  key="foo1"
  link="/c/default-cluster/ns/repo-namespace/charts/repo-name/foo"
  subIcon="helm.svg"
  tag1Class="repo-name"
  tag1Content={
    <ForwardRef
      className="ListItem__content__info_tag_link"
      to="/c/default-cluster/ns/repo-namespace/catalog/repo-name"
    >
      repo-name
    </ForwardRef>
  }
  title="foo"
>
  <Card
    className="ListItem"
    responsive={true}
  >
    <article
      className="Card ListItem Card-responsive"
    >
      <div
        className="Card__inner bg-white elevation-1"
      >
        <Link
          className="ListItem__header"
          title="foo"
          to="/c/default-cluster/ns/repo-namespace/charts/repo-name/foo"
        >
          <LinkAnchor
            className="ListItem__header"
            href="/c/default-cluster/ns/repo-namespace/charts/repo-name/foo"
            navigate={[Function]}
            title="foo"
          >
            <a
              className="ListItem__header"
              href="/c/default-cluster/ns/repo-namespace/charts/repo-name/foo"
              onClick={[Function]}
              title="foo"
            >
              <CardIcon
                icon="icon.png"
              >
                <div
                  className="Card__icon bg-light text-c"
                >
                  <img
                    alt="icon"
                    src="icon.png"
                  />
                </div>
              </CardIcon>
              <div
                className="Card__subIcon bg-light text-r"
              >
                <img
                  alt="icon"
                  src="helm.svg"
                />
              </div>
            </a>
          </LinkAnchor>
        </Link>
        <CardContent>
          <div
            className="Card__content padding-normal "
          >
            <div
              className="ListItem__content"
            >
              <Link
                title="foo"
                to="/c/default-cluster/ns/repo-namespace/charts/repo-name/foo"
              >
                <LinkAnchor
                  href="/c/default-cluster/ns/repo-namespace/charts/repo-name/foo"
                  navigate={[Function]}
                  title="foo"
                >
                  <a
                    href="/c/default-cluster/ns/repo-namespace/charts/repo-name/foo"
                    onClick={[Function]}
                    title="foo"
                  >
                    <h3
                      className="ListItem__content__title type-big"
                    >
                      foo
                    </h3>
                  </a>
                </LinkAnchor>
              </Link>
              <div
                className="ListItem__content__description"
              />
              <div
                className="ListItem__content__info"
              >
                <p
                  className="margin-reset type-small padding-t-tiny type-color-light-blue"
                >
                  1.0.0
                </p>
                <div>
                  <span
                    className="ListItem__content__info_tag ListItem__content__info_tag-1 type-small type-color-white padding-t-tiny padding-h-normal repo-name"
                  >
                    <Link
                      className="ListItem__content__info_tag_link"
                      to="/c/default-cluster/ns/repo-namespace/catalog/repo-name"
                    >
                      <LinkAnchor
                        className="ListItem__content__info_tag_link"
                        href="/c/default-cluster/ns/repo-namespace/catalog/repo-name"
                        navigate={[Function]}
                      >
                        <a
                          className="ListItem__content__info_tag_link"
                          href="/c/default-cluster/ns/repo-namespace/catalog/repo-name"
                          onClick={[Function]}
                        >
                          repo-name
                        </a>
                      </LinkAnchor>
                    </Link>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </CardContent>
      </div>
    </article>
  </Card>
</InfoCard>
`;

exports[`should render a global chart item in a namespace 1`] = `
<InfoCard
  description={
    <div
      className="ListItem__content__description"
    >
      
    </div>
  }
  icon="icon.png"
  info="1.0.0"
  key="foo1"
  link="/c/default-cluster/ns/repo-namespace/global-charts/repo-name/foo"
  subIcon="helm.svg"
  tag1Class="repo-name"
  tag1Content={
    <ForwardRef
      className="ListItem__content__info_tag_link"
      to="/c/default-cluster/ns/repo-namespace/catalog/repo-name"
    >
      repo-name
    </ForwardRef>
  }
  title="foo"
>
  <Card
    className="ListItem"
    responsive={true}
  >
    <article
      className="Card ListItem Card-responsive"
    >
      <div
        className="Card__inner bg-white elevation-1"
      >
        <Link
          className="ListItem__header"
          title="foo"
          to="/c/default-cluster/ns/repo-namespace/global-charts/repo-name/foo"
        >
          <LinkAnchor
            className="ListItem__header"
            href="/c/default-cluster/ns/repo-namespace/global-charts/repo-name/foo"
            navigate={[Function]}
            title="foo"
          >
            <a
              className="ListItem__header"
              href="/c/default-cluster/ns/repo-namespace/global-charts/repo-name/foo"
              onClick={[Function]}
              title="foo"
            >
              <CardIcon
                icon="icon.png"
              >
                <div
                  className="Card__icon bg-light text-c"
                >
                  <img
                    alt="icon"
                    src="icon.png"
                  />
                </div>
              </CardIcon>
              <div
                className="Card__subIcon bg-light text-r"
              >
                <img
                  alt="icon"
                  src="helm.svg"
                />
              </div>
            </a>
          </LinkAnchor>
        </Link>
        <CardContent>
          <div
            className="Card__content padding-normal "
          >
            <div
              className="ListItem__content"
            >
              <Link
                title="foo"
                to="/c/default-cluster/ns/repo-namespace/global-charts/repo-name/foo"
              >
                <LinkAnchor
                  href="/c/default-cluster/ns/repo-namespace/global-charts/repo-name/foo"
                  navigate={[Function]}
                  title="foo"
                >
                  <a
                    href="/c/default-cluster/ns/repo-namespace/global-charts/repo-name/foo"
                    onClick={[Function]}
                    title="foo"
                  >
                    <h3
                      className="ListItem__content__title type-big"
                    >
                      foo
                    </h3>
                  </a>
                </LinkAnchor>
              </Link>
              <div
                className="ListItem__content__description"
              />
              <div
                className="ListItem__content__info"
              >
                <p
                  className="margin-reset type-small padding-t-tiny type-color-light-blue"
                >
                  1.0.0
                </p>
                <div>
                  <span
                    className="ListItem__content__info_tag ListItem__content__info_tag-1 type-small type-color-white padding-t-tiny padding-h-normal repo-name"
                  >
                    <Link
                      className="ListItem__content__info_tag_link"
                      to="/c/default-cluster/ns/repo-namespace/catalog/repo-name"
                    >
                      <LinkAnchor
                        className="ListItem__content__info_tag_link"
                        href="/c/default-cluster/ns/repo-namespace/catalog/repo-name"
                        navigate={[Function]}
                      >
                        <a
                          className="ListItem__content__info_tag_link"
                          href="/c/default-cluster/ns/repo-namespace/catalog/repo-name"
                          onClick={[Function]}
                        >
                          repo-name
                        </a>
                      </LinkAnchor>
                    </Link>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </CardContent>
      </div>
    </article>
  </Card>
</InfoCard>
`;
